src/probability/kernel/cond_cdf.lean

```diff
@@ -0,0 +1,969 @@
+    { exact h_tendsto, }, },
+  classical,
+  let F : α → ℝ≥0∞ := λ a,
+    if h : ∃ l, tendsto (λ r, pre_cdf ρ (-r) a) at_top (𝓝 l) then h.some else 0,
+  have h_tendsto : ∀ᵐ a ∂ρ.fst, tendsto (λ r, pre_cdf ρ (-r) a) at_top (𝓝 (F a)),
+  { filter_upwards [h_exists] with a ha,
+    simp_rw [F, dif_pos ha],
+    exact ha.some_spec, },
+  suffices h_lintegral_eq : ∫⁻ a, F a ∂ρ.fst = 0,
+  { have hF_ae_meas : ae_measurable F ρ.fst,
+    { refine ae_measurable_of_tendsto_metrizable_ae _ (λ n, _) h_tendsto,
+      exact measurable_pre_cdf.ae_measurable, },
+    rw [lintegral_eq_zero_iff' hF_ae_meas] at h_lintegral_eq,
+    filter_upwards [h_tendsto, h_lintegral_eq] with a ha_tendsto ha_eq,
+    rwa ha_eq at ha_tendsto, },
+  have h_lintegral : tendsto (λ r, ∫⁻ a, pre_cdf ρ (-r) a ∂ρ.fst) at_top (𝓝 (∫⁻ a, F a ∂ρ.fst)),
+  { refine tendsto_lintegral_filter_of_dominated_convergence (λ _, 1)
+      (eventually_of_forall (λ _, measurable_pre_cdf)) (eventually_of_forall (λ _, _))
+      _ h_tendsto,
+    { filter_upwards [pre_cdf_le_one ρ] with a ha using ha _, },
+    { rw lintegral_one,
+      exact measure_ne_top _ _, }, },
+  have h_lintegral' : tendsto (λ r, ∫⁻ a, pre_cdf ρ (-r) a ∂ρ.fst) at_top (𝓝 0),
+  { have h_lintegral_eq : (λ r, ∫⁻ a, pre_cdf ρ (-r) a ∂ρ.fst) = λ r, ρ (univ ×ˢ Iic (-r)),
+    { ext1 n,
+      rw [← set_lintegral_univ, set_lintegral_pre_cdf_fst ρ _ measurable_set.univ,
+        measure.Iic_snd_univ],
+      norm_cast, },
+    rw h_lintegral_eq,
+    have h_zero_eq_measure_Inter : (0 : ℝ≥0∞) = ρ (⋂ r : ℚ, univ ×ˢ Iic (-r)),
+    { suffices : (⋂ r : ℚ, Iic (-(r : ℝ))) = ∅,
+      { rwa [← prod_Inter, this, prod_empty, measure_empty], },
+      ext1 x,
+      simp only [mem_Inter, mem_Iic, mem_empty_iff_false, iff_false, not_forall, not_le],
+      simp_rw neg_lt,
+      exact exists_rat_gt _, },
+    rw h_zero_eq_measure_Inter,
+    refine tendsto_measure_Inter (λ n, measurable_set.univ.prod measurable_set_Iic)
+      (λ i j hij x, _) ⟨0, measure_ne_top ρ _⟩,
+    simp only [mem_prod, mem_univ, mem_Iic, true_and],
+    refine λ hxj, hxj.trans (neg_le_neg _),
+    exact_mod_cast hij, },
+  exact tendsto_nhds_unique h_lintegral h_lintegral',
+end
+
+lemma inf_gt_pre_cdf (ρ : measure (α × ℝ)) [is_finite_measure ρ] :
+  ∀ᵐ a ∂ρ.fst, ∀ t : ℚ, (⨅ r : Ioi t, pre_cdf ρ r a) = pre_cdf ρ t a :=
+begin
+  rw ae_all_iff,
+  refine λ t, ae_eq_of_forall_set_lintegral_eq_of_sigma_finite _ measurable_pre_cdf _,
+  { exact measurable_infi (λ i, measurable_pre_cdf), },
+  intros s hs hs_fin,
+  rw [set_lintegral_infi_gt_pre_cdf ρ t hs, set_lintegral_pre_cdf_fst ρ t hs],
+end
+
+
+section has_cond_cdf
+
+/-- A product measure on `α × ℝ` is said to have a conditional cdf at `a : α` if `pre_cdf` is
+monotone with limit 0 at -∞ and 1 at +∞, and is right continuous.
+This property holds almost everywhere (see `has_cond_cdf_ae`). -/
+def has_cond_cdf (ρ : measure (α × ℝ)) (a : α) : Prop :=
```

Comment:
If instead of a big conjunction you defined it as a structure (still Prop-vaued) with several fields corresponding to each condition, this would be easier to read and to use.